Install GMT binary with BinDeps.jl

juliohm opened the pull request:
This PR uses the BinDeps.jl package to install the GMT binary when it is not installed already.

Feel free to add more providers for MacOS or a source provider to build a particular version of GMT from the sources.

joa-quim commented:
Thanks for this PR. I have no experience win BinDeps and I see that the PR uses Aptget. The problem with aptget is that the linux distros are awfully out of date and in ubuntu would install GMT5.2 which no longer supported by GMT.jl. Homebrew for Macs is OK. For Windows, I would like that the installer provided in the GMT downloads page is used (well, there currently no other options I think, but that will change in conda)

juliohm commented:
I have added the Homebrew provider for Mac as you suggested and removed the AptGet provider. I am also not very familiar with BinDeps, but it should work now with the latest GMT.

We can also have a fallback method that downloads the source and builds GMT locally as they do in CAIRO: https://github.com/JuliaGraphics/Cairo.jl/blob/master/deps/build.jl#L96-L125

Could you maybe share the build script for building GMT on the command line? Is it using GNU autotools, a raw Makefile, ...? I can try adapt it with BinDeps.jl

joa-quim commented:
Sorry for the delay. GMT builds with Cmake and instructions are [here](http://gmt.soest.hawaii.edu/projects/gmt/wiki/BuildingGMT) but sadly it misses the bit about the GDAL dependency, which I never remember the right name because I build on Windows almost all the times.

juliohm commented:
I think the safest option for now is to just merge the available providers in this PR and wait for future contributions. I don't have the time nor experience installing GMT myself. Someone familiar with the framework will likely provide better suggestions if he/she encounters errors in the build.

Do you want to merge this PR or should I close it?

joa-quim commented:
No, leave it open please. I don't have time to dedicate to this right now but I'll get back. We have also things developing on a Python side where a miniconda is being used to build the GMT trunk version. Certainly there can be an integration of that ... but later.

juliohm commented:
I will close this PR as it is showing up on my PR list as pending for a while. Please feel free to copy, modify, adapt the code here in any way you like in your own PR.

Best,

joa-quim commented:
OK.
